library/network_connections.py

```diff
@@ -2650,7 +2650,7 @@ def __init__(self, **kwargs):
     def run_prepare(self):
         Cmd.run_prepare(self)
         for idx, connection in enumerate(self.connections):
-            if connection['type'] in [ 'macvlan' ]:
+            if connection.get('type') in [ 'macvlan' ]:
```

Comment:
shouldn't it be `connection.get('type', None)` ?

Reply:
The default value is `None`.

Reply:
A, I didn't realize. I thought by default it throws an exception, that's why I often explicitly set `None` as default value.

Reply:
The exception happens when one uses `[]` instead of `.get()`.